### ISSUE TITLE
multiple copies of the gvmi image in their exe-unit/cache/tmp directory

### ISSUE THREAD

**Issue by mat7ias** [golemfactory/yagna#1319]
A provider machine can run out of disk space from having multiple copies of a gvmi image in their exe-unit/cache/tmp directory.
`(/root/.local/share/ya-provider/exe-unit/cache/tmp)` — potentially the solution could be to clear exe-unit/cache/tmp directory before starting a new task.

Relevant to Bug Bounty report #2: https://docs.google.com/document/d/1-ZyOOZ_f8g4qfLmcw0G1kEBxQCzALcZvTDc_6VUQSs4/edit?usp=sharing

**Comment by pnowosie** [comment]
I have some findings in the matter, I think it would be right to share in the related issue than opening the another.

When provider downloads an image file it stays in the `cache/tmp` directory and is also copied (with a name change) to `cache` directory. When a new agreement requiring the image is set, provider first check whether there is a file in the `cache/tmp` directory and then copies it over to the `cache` ignoring the file already exists in `cache` directory.
I think it was not the intended behaviour, which we might want to fix. **If we just clear the `cache/tmp` the image file will be downloaded from the registry despite we already have it in the `cache` **

**Comment by tworec** [comment]
closing via #1340
